File: 2704-maximum-difference-by-remapping-a-digit/2704-maximum-difference-by-remapping-a-digit.js
/**
 * @param {number} num
 * @return {number}
 */
var minMaxDifference = function(num) {
    const number1 = num.toString().split('');
    const number2 = num.toString().split('');

    let maxReplaced, maxIdx;
    for (let i = 0; i < number1.length; i++) {
        if (number1[i] !== '9') {
            maxReplaced = number1[i];
            maxIdx = i;
            number1[i] = '9';
            break;
        }
    }

    if (maxIdx !== undefined) {
        for (let i = maxIdx + 1; i < number1.length; i++) {
            if (number1[i] === maxReplaced) {
                number1[i] = '9';
            }
        }
    }

    let minReplaced, minIdx;
    for (let i = 0; i < number2.length; i++) {
        if (number2[i] !== '0') {
            minReplaced = number2[i];
            minIdx = i;
            number2[i] = '0';
            break;
        }
    }

    if (minIdx !== undefined) {
        for (let i = minIdx + 1; i < number2.length; i++) {
            if (number2[i] === minReplaced) {
                number2[i] = '0';
            }
        }
    }

    const ans = +number1.join('') - +number2.join('');
    return ans;
};

File: 2704-maximum-difference-by-remapping-a-digit/2704-maximum-difference-by-remapping-a-digit.cpp
class Solution {
public:
    int minMaxDifference(int num) {
        string number = to_string(num);
        int n = number.size();
        char maxReplace;
        int maxIdx = 0;
        for(int i=0; i<n; i++){
            if(number[i] != '9'){
                maxReplace = number[i];
                maxIdx = i;
                number[i] = '9';
                break;
            }
        }
        for(int i=maxIdx+1; i<n; i++){
            if(number[i] == maxReplace){
                number[i] = '9';
            }
        }
        string number2 = to_string(num);
        char minReplace;
        int minIdx = 0;
        for(int i=0; i<n; i++){
            if(number2[i] != '0'){
                minReplace = number2[i];
                minIdx = i;
                number2[i] = '0';
                break;
            }
        }
        for(int i=minIdx+1; i<n; i++){
            if(number2[i] == minReplace){
                number2[i] = '0';
            }
        }
        int num1 = stoi(number), num2 = stoi(number2);
        return num1-num2;
    }
};